webapp/src/ts/reducers/reports.ts

```diff
@@ -87,44 +95,52 @@ const _reportsReducer = createReducer(
     }
   ),
 
-  on(Actions.updateSelectedReportItem, (state, { payload }) => {
-    return {
-      ...state,
-      selected: state.selected?.map(item => {
-        if (item._id === payload.id) {
-          return { ...item, ...payload.selected };
-        }
-        return item;
-      }),
-    };
+  on(Actions.updateSelectedReportsItem, (state, { payload }) => {
+    let selectedReport = state.selectedReport;
+    if (selectedReport?._id === payload.id) {
+      selectedReport = { ...selectedReport, ...payload.selected };
+    }
+
+    const selectedReports = state.selectedReports?.map(item => {
+      return item._id === payload.id ? { ...item, ...payload.selected } : item;
+    });
+
+    return { ...state, selectedReport, selectedReports };
   }),
 
   on(Actions.setVerifyingReport, (state, { payload: { verifyingReport } }) => ({ ...state, verifyingReport })),
   on(Actions.toggleVerifyingReport, (state) => ({ ...state, verifyingReport: !state.verifyingReport })),
 
-  on(Actions.setFirstSelectedReportDocProperty, (state, { payload: { doc } }) => {
-    return {
-      ...state,
-      selected: state.selected?.map((item, idx) => {
-        if (idx === 0) {
-          return { ...item, doc: { ...item?.doc, ...doc } };
-        }
+  on(Actions.setSelectedReportDocProperty, (state, { payload: { doc } }) => {
+    if (!state.selectedReport) {
+      return state;
+    }
 
-        return item;
-      })
+    const selectedReport = {
```

Comment:
Can you also add the id check here as well, that you added in `updateSelectedReportsItem`. I have a feeling this will prevent race conditions where users click very quickly between reports.

Reply:
Good catch!! I'm going to return early if the `selectedReport._id` and `payload._id` are different
And will do the same for `setSelectedReportFormattedProperty`, with unit tests to validate the case